112.md

```diff
@@ -0,0 +1,248 @@
+}
+```
+
+The wrapped INNER event id for the channel creation should be saved.   This is the event id that will be used in all future INNER references to the channel creation.
+
+Channel creation events MUST be sent to the "destination pubkey" of the channel itself, not to any specific user.
+
+As a rule, this "destination pubkey" of the channel MUST used as the sole identifier and external filter that clients use to retrieve events for the channel.
+
+There are no other public tags or metadata, all other kinds and tags are "inner" (inside the wrapped message).  
+
+## Wrapped Kind 401: Invitation to Encrypted Channel
+ 
+Invitations are sent directly to the participants of the channel.
+
+```json
+{
+    "content": "optional personalized invitation message",
+    "tags": [["e", "channel-create-event-id", "relay-url"], ["privkey", "channel-private-key"]]
+}
+```
+
+When wrapping invitation messages, the destination pubkey is the user being invited, the relay used should be one of that user's well-known relays.  
+
+More than one relay can be recommended in the "e" tag.
+
+## Wrapped Kind 402: Change channel metadata
+
+Update an encrypted channel's public metadata.
+
+Clients and SHOULD handle wrapped-kind 402 events similar to kind 33 `replaceable` events.   Relays cannot distinguish this from other kind 1059 events.
+
+Clients SHOULD ignore kind wrapped-402 from pubkeys other than the current owner.
+
+The current owner of a channel is the pubkey that created the given channel using the corresponding wrapped-kind 400 event, or the current owner based on the 405-406 channel ownership delegation flow. 
+
+In other words, only the owner of a channel can *effectively* send kind wrapped-402 events; kind 402 for any channel sent by other people, even participants of that channel, are ignored.
+
+Clients SHOULD support basic metadata fields:
+
+- `name` - string - Channel name
+- `about` - string - Channel description
+- `picture` - string - URL of channel picture
+
+Clients MAY add additional metadata fields.
+
+Clients MUST specify an "e" tag to identify the channel id.
+Clients SHOULD mark the "e" tag to recommend a relay where the original event can be found.  
+Clients MAY replace the "r" tag set with a full list of new relays.
+
+There need be no overlap between the "e" relay and the "r" list.  For example, if a group's channel is moving its relay set to a new set.
+
+```json
+{
+    "content": "{\"name\": \"Updated Demo Channel\", \"about\": \"Updating a test channel.\", \"picture\": \"https://placekitten.com/201/201\"}",
+    "tags": [["e", "<channel-wrapped-400-event-id>", "wss://relay-url"]],
+}
+```
+
+
+## Kind 403: Send encrypted message to encrypted group channel
+
+Clients SHOULD use a marked "e" tags to specify if it is is a reply message.
+
+Clients SHOULD use [NIP-10](10.md) marked "e" tags to recommend a relay and specify whether it is a reply or root message.
+
+Clients SHOULD append [NIP-10](10.md) "p" tags to replies.
```

Comment:
```suggestion
Clients SHOULD use marked "e" tags to specify if it is a reply message.

Clients SHOULD use [NIP-10](10.md) marked "e" tags to recommend a relay and specify whether it is a reply or root message.

Clients SHOULD append [NIP-10](10.md) "p" tags to replies.
```

See above about encrypting tags. Leaking `e` tags is ok, but leaking `p` tags seems like a problem. Maybe group members wouldn't even want anyone to know they're members of a particular group.

Reply:
these tags are inside the gift wrap, so no problem with leaks.